pox/lib/addresses.py

```diff
@@ -387,14 +387,41 @@ def multicast_ethernet_address (self):
   def __str__ (self):
     return self.toStr()
 
-  def __cmp__ (self, other):
-    if other is None: return 1
+  def __eq__(self, other):
     try:
       if not isinstance(other, IPAddr):
         other = IPAddr(other)
-      return cmp(self.toUnsigned(), other.toUnsigned())
+      return self.toUnsigned() == other.toUnsigned()
     except:
-      return -other.__cmp__(self)
+      return other.__eq__(self)
+
+  def __ne__(self, other):
+    try:
+      if not isinstance(other, IPAddr):
+        other = IPAddr(other)
+      return self.toUnsigned() != other.toUnsigned()
+    except:
+      return other.__ne__(self)
+
+  def __lt__(self, other):
+    try:
+      if not isinstance(other, IPAddr):
+        other = IPAddr(other)
+      return self.toUnsigned() < other.toUnsigned()
+    except:
+      # reversed order
+      return other.__gt__(self) 
+
+  def __gt__(self, other):
+    try:
+      if not isinstance(other, IPAddr):
+        other = IPAddr(other)
+      return self.toUnsigned() > other.toUnsigned()
+    except:
+      # reversed order
+      return other.__lt__(self)
+
+
```

Comment:
Can we use the functools.total_ordering decorator here instead of a bunch of manual implementations?

Reply:
I guess so, but I believe this is a special case which I kept from the original code: we try to cast `other` to the same type as `self`, but if we fail, we do the reverse and call the compare function implemented in `other`.
I am not a fun of this logic, but I also do not know whether there were strong reason to do it this way in the first place.